api/handler/wasm.go

```diff
@@ -119,20 +119,20 @@ const (
 	// To allow downstream handlers to read the original request body, enable
 	// FeatureBufferRequest via FuncEnableFeatures. Otherwise, create a
 	// response inside the guest, or write an appropriate body via
-	// FuncSetRequestBody before calling FuncNext.
+	// FuncWriteRequestBody before calling FuncNext.
 	//
 	// TODO: document on http-wasm-abi
 	FuncReadRequestBody = "read_request_body"
 
-	// FuncSetRequestBody overwrites the request body with a value read from
-	// memory. In doing so, this overwrites the "Content-Length" header with
-	// its length.
+	// FuncWriteRequestBody writes up to `buf_len` bytes at memory offset `buf`
+	// to the body, and returns the length written.
```

Comment:
ps I'm tempted for us to not return the length written. rather just loop until that length is written host-side and panic if impossible. the client shouldn't need to loop.

Reply:
yeah, actually the impl raises an error if ever <n anyway, so there's no need to return len